projects/ngrx.io/content/guide/store/actions.md

```diff
@@ -85,6 +85,62 @@ The returned action has very specific context about where the action came from a
 
 </div>
 
+## Signal Actions
+
+There is also the option to dispatch a Signal of type `Signal<Action>`:
+
+<code-example header="book.component.ts">
+class BookComponent {
+  bookId = input.required&lt;number&gt;();
+  loadBookAction = computed(() => loadBook({ id: this.bookId() }));
+
+  constructor(store: Store) {
+    store.dispatch(this.loadBookAction);
+  }
+}
```

Comment:
Instead of creating a computed signal, wouldn't it be simpler to provide a callback (that will be used as `effectFn` under the hood) to the `dispatch` method:

```ts
```suggestion
class BookComponent {
  bookId = input.required&lt;number&gt;();

  constructor(store: Store) {
    this.store.dispatch(() => loadBook({ id: this.bookId() }));
  }
}
```

Reply:
I agree, but then I'd remove. the option to pass a Signal. I don't see any use in creating a Signal containing an action.

Reply:
>  I'd remove the option to pass a Signal

Never mind 😅. Signal is also just a function that returns an Action.

Reply:
Done, see comments below